### ISSUE TITLE
Do not allow character classes in switch case

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#1753]
#### For bugs
- Rule Id (if any, e.g. SC1000):
- My shellcheck version (`shellcheck --version` or "online"): online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

while read -r _spacecheck; do
	case ${_spacecheck} in
	line*:*)
	        echo "Match asterisk"
	        ;;
	*[[:space:]]:*)
                echo "Match character class"
		;;
	*)
		echo "No match"
	esac
done <<-EOT
	line one: no space
        line two : one space
        line three  : two spaces
EOT
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
While the bash manual says that character classes are allowed in case statements [0], at least in my default shell it does not work, so it seems like this behavior would be bash specific. Also my syntax highlighter is not happy.

[0] https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Pattern-Matching

**Comment by automorphism88** [comment]
Looks like your first pattern will match all three lines in your heredoc, so the second one is irrelevant and never gets evaluated. Why not test with just the one case pattern that's at issue? Another issue with your test script is that `read` is going to eliminate those extra spaces unless you add `IFS=` between while and read to prevent word splitting. Add an `echo "${_spacecheck}"` above the case statement and you'll see.

Maybe tell us what you expected the script's output to be, and what you got instead that makes you think shellcheck should flag it? Or if it's a bash vs sh issue, have you tested it in bash? And what is your default shell? It looks to me like POSIX also specifies character classes are allowed in case patterns (https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html), and if your shell is not POSIX sh compliant that's a bug in the shell, not shellcheck.
